### PR TITLE
Fix arcade build error 

### PR DESCRIPTION
**Pull request by epananth**
Fix for => https://github.com/dotnet/arcade/issues/5995

test build that passed => https://dnceng.visualstudio.com/internal/_build/results?buildId=778722&view=results

PublishingInfraVersion(for V3) has mandatory fields in 2 places 

1) eng/Publishing.props
2) post-build.yml

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
     <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
     <add key="vs-buildservices" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json" />
+    <add key="general-testing" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/general-testing/nuget/v3/index.json" />
```
Comment (epananth) [inline NuGet.config:12]: will revert this after the build turns green

File: eng/Publishing.props [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<Project>
```
Comment (riarenas) [inline eng/Publishing.props:1]: [nit] this file needs proper indentation, and the xml declaration

File: eng/common/templates/post-build/post-build.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ parameters:
  # Which publishing infra should be used. THIS SHOULD MATCH THE VERSION ON THE BUILD MANIFEST.
   # Publishing V2 accepts optionally outlining the publishing stages - default is inline.
   # Publishing V3 DOES NOT accept inlining the publishing stages.
-  publishingInfraVersion: 2
+  publishingInfraVersion: 3
```
Comment (riarenas) [inline eng/common/templates/post-build/post-build.yml:5]: Why are we setting the default to 3?

File: NuGet.config (hunk 1 above)
Comment (riarenas) [reply NuGet.config:12]: I don't even think you need this for tests.

You would only need this if you were restoring something from a build that published to General testing. Doesn't seem to be the case here

File: eng/common/post-build/publish-using-darc.ps1 [hunk 4]
```diff
@@ -54,7 +54,7 @@ try {
   --id $buildId `
   --publishing-infra-version $PublishingInfraVersion `
   --default-channels `
-  --source-branch master `
```
Comment (riarenas) [inline eng/common/post-build/publish-using-darc.ps1:57]: Why this change? even for testing I'm pretty sure you want this to use master?

File: NuGet.config (hunk 1 above)
Comment (epananth) [reply NuGet.config]: reverted this one, this is not necessary..

File: eng/common/post-build/publish-using-darc.ps1 (hunk 4 above)
Comment (epananth) [reply eng/common/post-build/publish-using-darc.ps1]: reverting this too. In this case master is same as my test branch,I reverted this too..

File: eng/common/templates/post-build/post-build.yml (hunk 3 above)
Comment (epananth) [reply eng/common/templates/post-build/post-build.yml]: I wanted to test for v3, but I don't have to..  I will keep this as I will merge to master. Sorry for the confusion

**Comment by epananth** [comment]
> The test build didn't run any publishing
> 
> ```
> 'D:\a\1\s\.dotnet\dotnet.exe' tool install microsoft.dotnet.darc --version 1.1.0-beta.20418.1 --add-source 'https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json' -v minimal -g
>   Determining projects to restore...
>   Restored C:\Users\VssAdministrator\AppData\Local\Temp\s3gyipsg.f03\restore.csproj (in 5.41 sec).
> You can invoke the tool using the following command: darc
> Tool 'microsoft.dotnet.darc' (version '1.1.0-beta.20418.1') was successfully installed.
> Build '61036' is already on all target channel(s).
> ```

Since I used a github branch I need to add-default-channel using github repo url. I did that triggered a build.

File: eng/common/templates/post-build/post-build.yml (hunk 3 above)
Comment (riarenas) [reply eng/common/templates/post-build/post-build.yml]: This would've changed the default that the template uses if a value is not passed (every repo except arcade right now). This wasn't necessary to test version 3 as Arcade is supposedly already passing that in through https://github.com/dotnet/arcade/blob/aab26072c0bf65a858935cf7ace036a2d615e38f/azure-pipelines.yml#L216

File: eng/Publishing.props (hunk 2 above)
Comment (riarenas) [reply eng/Publishing.props:1]: Please add the xml declaration

Comment (riarenas) [inline eng/Publishing.props:1]:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<Project>
```

**Comment by epananth** [comment]
> I'm confused again, why are we saying publishing version 2 in Publishing.props?

was testing something else, reverted that
